### ISSUE TITLE
Security of Firestore credentials

### ISSUE THREAD

**Issue by jonmountjoy** [JagTheFriend/Nextjs-Blog#1]
Thanks for this!

Given that all the credentials are `NEXT_PUBLIC` (ie. `NEXT_PUBLIC_FIREBASE_API_KEY`), does this mean that anyone can read all the credentials client-side, and then build their own queries and apps against your database (and potentially abuse it in whatever way)?

**Comment by JagTheFriend** [comment]
> Given that all the credentials are `NEXT_PUBLIC` (ie. `NEXT_PUBLIC_FIREBASE_API_KEY`), does this mean that anyone can read all the credentials client-side, and then build their own queries and apps against your database (and potentially abuse it in whatever way)?

No, from what I have heard, google wouldn't allow everyone to use other's firebase API key.
Since those (important) API key would be exposed while being used in the frontend anyways.
